From 6ae84f1183e91c910ca92a55e37f8254ace805c0 Mon Sep 17 00:00:00 2001
From: Jiang Jiang <jiang.jiang@nokia.com>
Date: Mon, 6 Dec 2010 13:07:36 +0100
Subject: [PATCH] Fix QTextEdit::selectAll crash from textChanged()

Doing selectAll() after the entire block of text has been removed
will cause this crash, because we didn't check if the block we
found is valid or not.

Task-number: QTBUG-15857
Reviewed-by: Eskil
---
 src/gui/text/qtextcursor.cpp |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

Index: qt-everywhere-opensource-src-4.7.2/src/gui/text/qtextcursor.cpp
===================================================================
--- qt-everywhere-opensource-src-4.7.2.orig/src/gui/text/qtextcursor.cpp	2011-02-22 12:03:59.000000000 +0000
+++ qt-everywhere-opensource-src-4.7.2/src/gui/text/qtextcursor.cpp	2011-03-02 18:09:49.157309610 +0000
@@ -366,6 +366,9 @@
     if (!blockIt.isValid())
         return false;
 
+    if (!blockIt.isValid())
+        return false;
+
     if (op >= QTextCursor::Left && op <= QTextCursor::WordRight
         && blockIt.textDirection() == Qt::RightToLeft) {
         if (op == QTextCursor::Left)
